nf_core/lint.py

```diff
@@ -729,9 +666,9 @@ def check_version_consistency(self):
         if self.config.get('process.container', ''):
             versions['process.container'] = self.config.get('process.container', '').strip(' \'"').split(':')[-1]
 
-        # Get version from the TRAVIS_TAG env var
-        if os.environ.get('TRAVIS_TAG') and os.environ.get('TRAVIS_REPO_SLUG', '') != 'nf-core/tools':
-            versions['TRAVIS_TAG'] = os.environ.get('TRAVIS_TAG').strip(' \'"')
+        # Get version from the GITHUB_REF env var
+        if os.environ.get('GITHUB_REF') and os.environ.get('GITHUB_REPOSITORY', '') != 'nf-core/tools':
```

Comment:
So one problem here is that the linting currently doesn't run for releases - only pushes and PRs:
https://github.com/nf-core/tools/blob/25335b387589008838c83fb0a73808cc0f9a7cbb/nf_core/pipeline-template/%7B%7Bcookiecutter.name_noslash%7D%7D/.github/workflows/linting.yml#L1-L4

So as a first step, we need to add `release` there (see [GitHub Actions events docs](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#release-event-release)):

``` yaml
on:
  release:
    types: [published]
```

Slightly unclear how to combine this what we have at the moment, I guess this? ([docs](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#example-using-multiple-events-with-activity-types-or-configuration))

```yaml
on:
  push:
  pull_request:
  release:
    types: [published]
```

Reply:
`GITHUB_REF` isn't specific to releases, so we need to check that it begins with `refs/tags/` to limit it to releases, instead of just checking if it is set:

```suggestion
        if os.environ.get('GITHUB_REF', '').startswith('refs/tags/') and os.environ.get('GITHUB_REPOSITORY', '') != 'nf-core/tools':
```

Reply:
See https://github.com/nf-core/tools/pull/502/commits/8fce2b619791e8c0f151230bf6c3987fe63bb97b